### DISCUSSION TITLE
Color citation content but not brackets

### DISCUSSION THREAD

**Discussion by TheZoq2** [typst/typst#4143] [Q&A] [answered]
Several papers add colors to the citation numbers, for example:
![image](https://github.com/typst/typst/assets/5312813/b438301c-a8f2-4c19-9ece-f60ec3c994a5)

I'm trying to get the same effect in typst, but I can't figure out how. The closest I've got is with `#show cite: set text(fill: purple)`, but that also colors the 
![image](https://github.com/typst/typst/assets/5312813/19456820-efed-42d0-b799-55383e7ec846)
brackets around the numbers which I don't want.

I found some old threads here that say you can `show link: set text(fill: purple)` or similar, but that has no effect.

Is there a convenient way to do this?

**Comment by EpicEricEE** [comment, marked as answer]
Something like this should work. As `cite` elements are also references, you need to exclude them when coloring other references, so you don't end up with differently colored brackets:

```typ
#show cite: it => {
  // Only color the number, not the brackets.
  show regex("\d+"): set text(fill: green)
  // or regex("[\p{L}\d+]+") when using the alpha-numerical style
  it
}

#show ref: it => {
  if it.element == none {
    // This is a citation, which is handled above.
    return it
  }

  // Only color the number, not the supplement.
  show regex("[\d.]+"): set text(fill: blue)
  it
}
```

![image](https://github.com/typst/typst/assets/7191192/ff690cbf-a42b-4698-89e9-8184f4c5bb22)

...or alpha-numerical with `regex("[\p{L}\d+]+")`
![image](https://github.com/typst/typst/assets/7191192/df0a78af-119c-4c71-8d09-4dadb0bcf124)

**Reply by TheZoq2** [reply]
Oh using regex there is clever! Thanks
